kafka-rest/src/test/java/io/confluent/kafkarest/integration/v3/AllTopicsConfigsResourceIntegrationTest.java

```diff
@@ -0,0 +1,297 @@
+/*
+ * Copyright 2020 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.kafkarest.integration.v3;
+
+import io.confluent.kafkarest.entities.ConfigSource;
+import io.confluent.kafkarest.entities.v3.ConfigSynonymData;
+import io.confluent.kafkarest.entities.v3.ListTopicConfigsResponse;
+import io.confluent.kafkarest.entities.v3.Resource;
+import io.confluent.kafkarest.entities.v3.ResourceCollection;
+import io.confluent.kafkarest.entities.v3.TopicConfigData;
+import io.confluent.kafkarest.integration.ClusterTestHarness;
+import org.junit.Before;
+import org.junit.Test;
+
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.Response.Status;
+
+import java.util.Arrays;
+
+import static java.util.Collections.singletonList;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+public class AllTopicsConfigsResourceIntegrationTest extends ClusterTestHarness {
+
+    private static final String TOPIC_1 = "topic-1";
+    private static final String TOPIC_2 = "topic-2";
+
+    public AllTopicsConfigsResourceIntegrationTest() {
+        super(/* numBrokers= */ 1, /* withSchemaRegistry= */ false);
+    }
+
+    @Before
+    @Override
+    public void setUp() throws Exception {
+        super.setUp();
+
+        createTopic(TOPIC_1, 1, (short) 1);
+        createTopic(TOPIC_2, 1, (short) 1);
+        setTopicConfig(TOPIC_2,"delete.retention.ms","100000");
+
+    }
+
+    @Test
+    public void listTopicConfigs_existingTopics_returnsConfigs() {
```

Comment:
nit: Wouldn't introducing utility methods for creating the cleanup policy config, the compression type config, and the retention config cut down the verbosity of this test?

Reply:
yep, I added a config generation utility method.